examples/retinanet/util.py

```diff
@@ -0,0 +1,506 @@
+from jax import numpy as jnp
+
+import jax
+import numpy as np
+import tensorflow as tf
+
+CATEGORY_MAP = {
+    0: "background",
+    1: "person",
+    2: "bicycle",
+    3: "car",
+    4: "motorcycle",
+    5: "airplane",
+    6: "bus",
+    7: "train",
+    8: "truck",
+    9: "boat",
+    10: "traffic light",
+    11: "fire hydrant",
+    12: "stop sign",
+    13: "parking meter",
+    14: "bench",
+    15: "bird",
+    16: "cat",
+    17: "dog",
+    18: "horse",
+    19: "sheep",
+    20: "cow",
+    21: "elephant",
+    22: "bear",
+    23: "zebra",
+    24: "giraffe",
+    25: "backpack",
+    26: "umbrella",
+    27: "handbag",
+    28: "tie",
+    29: "suitcase",
+    30: "frisbee",
+    31: "skis",
+    32: "snowboard",
+    33: "sports ball",
+    34: "kite",
+    35: "baseball bat",
+    36: "baseball glove",
+    37: "skateboard",
+    38: "surfboard",
+    39: "tennis racket",
+    40: "bottle",
+    41: "wine glass",
+    42: "cup",
+    43: "fork",
+    44: "knife",
+    45: "spoon",
+    46: "bowl",
+    47: "banana",
+    48: "apple",
+    49: "sandwich",
+    50: "orange",
+    51: "broccoli",
+    52: "carrot",
+    53: "hot dog",
+    54: "pizza",
+    55: "donut",
+    56: "cake",
+    57: "chair",
+    58: "couch",
+    59: "potted plant",
+    60: "bed",
+    61: "dining table",
+    62: "toilet",
+    63: "tv",
+    64: "laptop",
+    65: "mouse",
+    66: "remote",
+    67: "keyboard",
+    68: "cell phone",
+    69: "microwave",
+    70: "oven",
+    71: "toaster",
+    72: "sink",
+    73: "refrigerator",
+    74: "book",
+    75: "clock",
+    76: "vase",
+    77: "scissors",
+    78: "teddy bear",
+    79: "hair drier",
+    80: "toothbrush"
+}
+
+
+def pi_init(pi):
+  """Wrapper to log-based weight initializer function.
+
+  This initializer is used for the bias term in the classification subnet, as
+  described in https://arxiv.org/abs/1708.02002
+
+  Args:
+    pi: the prior probability of detecting an object
+
+  Returns:
+    An array for initializing a module's weights / biases
+  """
+
+  def _inner(key, shape, dtype=jnp.float32):
+    return jnp.ones(shape, dtype) * (-jnp.log((1 - pi) / pi))
+
+  return _inner
+
+
+@jax.vmap
+def clip_anchors(anchors, height, width):
```

Comment:
Seems very similar to `anchors.clip_anchors()`

Reply:
Indeed. The difference between the two is that one works on a batch, and the other at the image level. I can try to remove one (the vmapped one), and only use the other in `anchors`.